### ISSUE TITLE
Schema ignores mandatory key constraint from super

### ISSUE THREAD

**Issue by soliton-** [wesnoth/wesnoth#7897]
Given WML schema test_schema.cfg:
```
[wml_schema]
    [type]
        name=id
        value="[a-zA-Z0-9_~$]+"
    [/type]
    [tag]
        name="root"
        min=1
        [tag]
            name="virtual~super"
            max=0
            [key]
                name=id
                type=id
                mandatory=true
            [/key]
        [/tag]
        [tag]
            name="mytag"
            min=1
            [tag]
                name="campaign"
                max=infinite
                super="virtual~super"
            [/tag]
        [/tag]
    [/tag]
[/wml_schema]
```

and test_schema_test.cfg:
```
[mytag]
    [campaign]
#        id=*
    [/campaign]
[/mytag]
```

`wesnoth -S test_schema.cfg --validate test_schema_test.cfg` outputs:
```
Validating test_schema_test.cfg against schema /home/USER/test_schema.cfg
validation succeeded
```

So the mandatory=true part is ignored. While the type part is not ignored as can be seen when uncommenting the id line.
```
error validation: Invalid value '*' in key 'id=' in tag [campaign]
 (expected value of type id) at test_schema_test.cfg:3
```

**Comment by soliton-** [comment]
Also https://github.com/wesnoth/wesnoth/blob/e7252b212e4a1b6f92b9cbe8be67eef7295f112b/src/serialization/schema/tag.cpp#L367
should *really* be done. Any message is better than nothing.

**Comment by soliton-** [comment]
I think we should try to get the schema as accurate as possible for 1.18.

**Comment by Pentarctagon** [comment]
@CelticMinstrel can you look into this?

**Comment by CelticMinstrel** [comment]
I don't think I can make any promises right now. If you're asking whether I can do it, the answer is of course yes, but it might not be until next year. I can't really predict a timeline.

**Comment by rfsc-mori** [comment]
I've decided to try to fix to learn more about the project, and after a few days learning how the preprocessor, parser and schema validator worked I found a solution I was satisfied with. And then decided to open the PR in the case it was good enough fix. I apologize if I'm getting ahead of myself (since CelticMinstrel would be the one to look into this later on). I hope to not have caused two or more people to work on the same issue.
